hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/utils/HawkbitCommonUtil.java

```diff
@@ -596,7 +596,7 @@ public static Locale getLocaleToBeUsed(final UiProperties.Localization localizat
      */
     public static void initLocalization(final UI ui, final Localization localizationProperties,
             final VaadinMessageSource i18n) {
-        ui.setLocale(HawkbitCommonUtil.getLocaleToBeUsed(localizationProperties, ui.getSession().getLocale()));
+        ui.setLocale(getLocaleToBeUsed(localizationProperties, Page.getCurrent().getWebBrowser().getLocale()));
```

Comment:
do we really need to get the current page to get the desired locale?

Reply:
You can use `ui.getPage()` instead of `Page.getCurrent()` here, not to search for the UI one more time.